src/IssueViz/Editor/LocationNavigator.cs

```diff
@@ -47,33 +49,43 @@ internal LocationNavigator(IDocumentNavigator documentNavigator, IIssueSpanCalcu
             this.logger = logger;
         }
 
-        public bool TryNavigate(IAnalysisIssueLocation location)
+        public bool TryNavigate(IAnalysisIssueLocationVisualization locationVisualization)
         {
-            var locationFilePath = location?.FilePath;
-
-            if (string.IsNullOrEmpty(locationFilePath))
+            if (locationVisualization == null)
             {
-                return false;
+                throw new ArgumentNullException(nameof(locationVisualization));
             }
 
             try
             {
-                var textView = documentNavigator.Open(locationFilePath);
-                var locationSpan = spanCalculator.CalculateSpan(location, textView.TextBuffer.CurrentSnapshot);
+                var textView = documentNavigator.Open(locationVisualization.CurrentFilePath);
+                var locationSpan = GetLocationSpan(locationVisualization, textView);
 
-                if (locationSpan.HasValue)
+                if (!locationSpan.IsEmpty)
                 {
-                    documentNavigator.Navigate(textView, locationSpan.Value);
+                    documentNavigator.Navigate(textView, locationSpan);
 
                     return true;
                 }
             }
             catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
             {
-                logger.WriteLine(Resources.ERR_OpenDocumentException, locationFilePath, ex.Message);
+                logger.WriteLine(Resources.ERR_OpenDocumentException, locationVisualization.CurrentFilePath, ex.Message);
             }
 
             return false;
         }
+
+        private SnapshotSpan GetLocationSpan(IAnalysisIssueLocationVisualization locationVisualization, ITextView textView)
+        {
+            var shouldCalculateSpan = !locationVisualization.Span.HasValue || locationVisualization.Span.Value.IsEmpty;
+
+            if (shouldCalculateSpan)
```

Comment:
I'm debating about `shouldCalculateSpan`.
If the user navigates to the location once (thus setting Span to a valid value), and then modifies the code without triggering the tagger (e.g. outside of VS) -- we will never reset the wrong span and will keep trying navigating to it.

Reply:
What's the alternative? Always calculating the span? That would be the wrong thing to do in any case where the user has edited the document inside VS so the tagger will have updated `locationVisualization.Span`.